osu.Game/Beatmaps/Formats/LegacyDecoder.cs

```diff
@@ -40,7 +41,53 @@ protected override void ParseStreamInto(StreamReader stream, T beatmap)
 
         protected virtual bool ShouldSkipLine(string line) => string.IsNullOrWhiteSpace(line) || line.StartsWith("//");
 
-        protected abstract void ParseLine(T output, Section section, string line);
+        protected virtual void ParseLine(T output, Section section, string line)
+        {
+            switch (section)
+            {
+                case Section.Colours:
+                    handleColours(output, line);
+                    return;
+            }
+        }
+
+        private bool hasCustomColours;
+
+        private void handleColours(T output, string line)
+        {
+            var pair = SplitKeyVal(line, ':');
+
+            bool isCombo = pair.Key.StartsWith(@"Combo");
+
+            string[] split = pair.Value.Split(',');
+
+            if (split.Length != 3)
+                throw new InvalidOperationException($@"Color specified in incorrect format (should be R,G,B): {pair.Value}");
+
+            if (!byte.TryParse(split[0], out var r) || !byte.TryParse(split[1], out var g) || !byte.TryParse(split[2], out var b))

```

Comment:
shouldn't we be trimming each of these?

Reply:
They aren't trimmed in stable, so I think it's safe to not do so